README.md

```diff
@@ -114,6 +114,22 @@ It is possible to generate spdx based SBOM (software bill of materials)
 documents as part of the build. Just pass `--with-sbom` and
 it will put them into the output directory.
 
+### Cloud integration
+
+When building an image type that can be uploaded to the cloud
+(e.g. an "ami") image-builder will automatically upload if
+all cloud paramters are provided, e.g.
+```
+$ imge-builder build ami --distro centos-9 \
+    --aws-region=us-east-1 \
+	--aws-bucket=example-bucket \
+	--aws-ami-name=my-image-1
+```
+Images can also be uploaded with the `image-builder upload` command
+after they are build.
```

Comment:
This really calls for an explicit `--upload` option for the `ami` argument, just providing AWS credentialy should not automatically upload I think.

Reply:
Thanks, would love to hear what @achilleas-k and @ondrejbudai think here, I mostly followed what bib is doing but if this is actually making your life harder and/or something that (most?) users would not expect we should add an explicit upload option. Looking forward for comments :)

Reply:
@ondrejbudai one thing to consider is what I wrote in https://github.com/osbuild/image-builder-cli/pull/119#discussion_r1942524877 - i.e. when we do this implicitly when we have all parameters for an upload we cannot show a nice error if a parameter is missing. That would be a reason for the explicit flag because then we could error if anything is missing.

Reply:
I do not have really any logical argument for liking explicit other than it feels more natural to me when there is a specific command to do the same thing:

```
git rebase
git pull --rebase
```

Maybe one argument, it is very common for AWS client tools to accept these values as environment variables, in fact, `bootc-image-builder` does this too. How one can upload an image then all options are specified this way? In the AWS case, there is an option that is required and must be passed regularly, but there might be other clouds that do not have such option. It could be as easy as:

```
export ORACLE_CLOUD_CREDENTIALS=creds.json
cli build --and-upload-to oracle
```